src/main/java/io/kestra/plugin/notifications/sendgrid/SendGridMailTemplate.java

```diff
@@ -44,7 +54,15 @@ public SendGridMailSend.Output run(RunContext runContext) throws Exception {
             );
         }
 
+        if (this.textTemplateUri != null) {
+            plainTextTemplate = IOUtils.toString(
+                Objects.requireNonNull(this.getClass().getClassLoader().getResourceAsStream(this.textTemplateUri)),
+                Charsets.UTF_8
+            );
+        }
+
         this.htmlContent = runContext.render(htmlTextTemplate, templateRenderMap != null ? templateRenderMap : Map.of());
```

Comment:
@tchiotludo any specific reason to not using here `
```
this.htmlContent = runContext.render(htmlTextTemplate, io.kestra.core.utils.MapUtils.emptyOnNull(templateRenderMap));

```

Reply:
You can use it here, good idea.

Reply:
@loicmathieu not in scope of this PR, but other templates all have the same checks and maybe as a part of refactoring in separate PR can be done that.

Reply:
Handled https://github.com/kestra-io/plugin-notifications/pull/139/commits/d84fa92be782dbdafc72662731f539022174ba06